libs/sdk-ui-charts/src/highcharts/chartTypes/warterfallChart/waterfallConfiguration.ts

```diff
@@ -0,0 +1,45 @@
+// (C) 2023 GoodData Corporation
+import { IChartConfig } from "../../../interfaces";
+import { HighchartsOptions } from "../../lib";
+import { MAX_POINT_WIDTH } from "../_chartCreators/commonConfiguration";
+import { getCommonResponsiveConfig } from "../_chartCreators/responsive";
+import { getAxesCounts } from "../_util/common";
+
+export function getWaterfallConfiguration(config: IChartConfig): HighchartsOptions {
+    const waterfallConfiguration = {
+        chart: {
+            type: "waterfall",
+        },
+        plotOptions: {
+            waterfall: {
+                maxPointWidth: MAX_POINT_WIDTH,
+                dataLabels: {
+                    enabled: true,
+                    padding: 2,
+                    verticalAlign: "top",
+                    y: -20,
+                },
+            },
+            series: {
+                states: {
+                    hover: {
+                        enabled: false,
+                    },
+                },
+            },
+        },
+        legend: {
+            enabled: false,
+        },
+    };
+
+    if (config?.enableCompactSize) {
+        const [xAxesCount, yAxesCount] = getAxesCounts(config);
+        return {
+            ...waterfallConfiguration,
+            responsive: getCommonResponsiveConfig(false, xAxesCount, yAxesCount),
+        };
+    }
+
+    return waterfallConfiguration;
```

Comment:
This is matter of preference, but it is better to switch the conditions. Define default object, if it is not compact size return it, the you can do the modification that you will return. 

````
if (!config?.enableCompactSize) {
   return waterfallConfiguration;
}

const [xAxesCount, yAxesCount] = getAxesCounts(config);
return {
   ...waterfallConfiguration,
   responsive: getCommonResponsiveConfig(false, xAxesCount, yAxesCount),
};
````

Reply:
I updated this. Thank you!